celery/backends/amqp.py

```diff
@@ -113,12 +113,24 @@ def store_result(self, task_id, result, state,
         routing_key, correlation_id = self.destination_for(task_id, request)
         if not routing_key:
             return
+
+        payload = {'task_id': task_id, 'status': state,
+                   'result': self.encode_result(result, state),
+                   'traceback': traceback,
+                   'children': self.current_task_children(request)}
+        if self.app.conf.find_value_for_key('extended', 'result'):
+            payload['name'] = getattr(request, 'task_name', None)
```

Comment:
I think the task name attribute is `task` and not `task_name`, see the pull request #5486.